### PR TITLE
OPCUA-717 improve uavariant and uastring

### PR DESCRIPTION
**Pull request by pnikiel**
1. UaString impl now as a separate cpp file
2. UaVariant impl now as a separate cpp file
3. New implementation for UaVariant::type() avoiding longish and multiple if.. else.. if.. else.. and profiting from already existing implementation in open62541

### CODE REVIEW

**Review by ben-farnham — COMMENTED** [review]
Hi Piotr,

A general question - this a reorganisation of code (rather than 'new logic')? It's a bit hard to tell from the github diff view.

Cheers,
Ben

**Comment by ben-farnham** [comment]
Ooops, once again - please ignore my previous comment; just spotted your description of what it is at the top.

Right, this time I'm concentrating...

**Comment by pnikiel** [comment]
It is reorganisation AND some new code for UaVariant::type().

cheers p

File: src/uavariant.cpp [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+}
+
+UaVariant::~UaVariant()
+{
+    LOG(Log::TRC) <<"+"<< __PRETTY_FUNCTION__ << " m_impl="<<m_impl<<" m_impl.data="<<m_impl->data;
+    destroyOpen62541Variant(m_impl);
+    m_impl = 0;
+}
+
+OpcUaType UaVariant::type() const
```
Comment (ben-farnham) [inline src/uavariant.cpp:146]: Nice!

**Review by ben-farnham — APPROVED** [review]
Looks good I think. Much prefer the new UaVariant::type() implementation.

**Comment by pnikiel** [comment]
Thanks Ben!
Michael, do you plan to review or we get a blind green light?

**Comment by meeludwig** [comment]
A blind green light from me I am afraid, concerning quasar UaVariant updates.
Michael

**Comment by pnikiel** [comment]
OK!

**Pull request merged**